### PR TITLE
Modify: score_at_threshold where query with date ranges style

### PR DESCRIPTION
**Pull request by roharon**
What about use data range style instead of defining comparative conditions on `where` query?

### CODE REVIEW

**Comment by ClearlyClaire** [comment]
We're still targeting Ruby 2.6 for now, which doesn't have support for this notation!

File: app/models/trends/links.rb
Comment (casaper) [inline app/models/trends/links.rb:76]: Suggest code style fix, when you touch it for such a cosmetic change:

```rb
score_at_threshold = PreviewCardTrend.order(rank: :desc)
                                     .where(language: language, allowed: true, rank: ..options[:review_threshold])
                                     .first&.score || 0
```

**Review by casaper — CHANGES_REQUESTED** [review]

**Comment by roharon** [comment]
> We're still targeting Ruby 2.6 for now, which doesn't have support for this notation!

Oops I missed it!
Thanks for comment.

**Comment by ClearlyClaire** [comment]
> > We're still targeting Ruby 2.6 for now, which doesn't have support for this notation!
> 
> Oops I missed it! Thanks for comment.

It appears we recently inadvertently dropped support for Ruby 2.6 anyway, so we're going to formally drop compatibility soon and update rubocop's config accordingly. So this PR is blocked by #21477 but can be merged right after.

**Comment by roharon** [comment]
Okay 😅  I m gonna check Lint Code Base status again after ruby parser updated (maybe seems like after https://github.com/mastodon/mastodon/pull/21477 merged)

**Comment by ClearlyClaire** [comment]
We've officially dropped 2.6 support, you can now rebase your change!

**Comment by roharon** [comment]
@ClearlyClaire Head repository has been synced changes!

**Review by ClearlyClaire — APPROVED** [review]

**Comment by Gargron** [comment]
I think this style is much less readable?

**Review by mjankowski — APPROVED** [review]

**Pull request closed without merging**
